shopfloor/models/stock_move.py

```diff
@@ -18,13 +18,22 @@ def split_other_move_lines(self, move_lines, intersection=False):
             other_move_lines = self.move_line_ids & move_lines
         else:
             other_move_lines = self.move_line_ids - move_lines
-        if other_move_lines:
-            qty_to_split = sum(other_move_lines.mapped("product_uom_qty"))
+        if other_move_lines or self.state == "partially_available":
+            if intersection:
+                # TODO @sebalix: please check if we can abandon the flag.
```

Comment:
Ok I see, indeed better to do the "set operation" outside of this method. Can't see a side-effect by doing so.